x/ccv/provider/types/msg.go

```diff
@@ -388,352 +374,54 @@ func NewMsgRemoveConsumer(signer string, consumerId string, stopTime time.Time)
-		}
-	}
-	return nil
-}
-
-// NewMsgOptOut creates a new NewMsgOptIn instance.
-func NewMsgOptOut(consumerId string, providerValidatorAddress sdk.ValAddress, signer string) (*MsgOptOut, error) {
-	return &MsgOptOut{
-		ConsumerId:   consumerId,
-		ProviderAddr: providerValidatorAddress.String(),
-		Signer:       signer,
-	}, nil
-}
-
-// Type implements the sdk.Msg interface.
-func (msg MsgOptOut) Type() string {
-	return TypeMsgOptOut
-}
-
-// Route implements the sdk.Msg interface.
-func (msg MsgOptOut) Route() string { return RouterKey }
-
-// GetSigners implements the sdk.Msg interface. It returns the address(es) that
-// must sign over msg.GetSignBytes().
-func (msg MsgOptOut) GetSigners() []sdk.AccAddress {
-	valAddr, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
-	if err != nil {
-		// same behavior as in cosmos-sdk
-		panic(err)
-	}
-	return []sdk.AccAddress{valAddr.Bytes()}
-}
-
-// GetSignBytes returns the message bytes to sign over.
-func (msg MsgOptOut) GetSignBytes() []byte {
-	bz := ccvtypes.ModuleCdc.MustMarshalJSON(&msg)
-	return sdk.MustSortJSON(bz)
-}
-
-// ValidateBasic implements the sdk.Msg interface.
-func (msg MsgOptOut) ValidateBasic() error {
-	if err := ValidateConsumerId(msg.ConsumerId); err != nil {
-		return err
+// ParseConsumerKeyFromJson parses the consumer key from a JSON string,
+// this replaces deserializing a protobuf any.
+func ParseConsumerKeyFromJson(jsonStr string) (pkType, key string, err error) {
+	type PubKey struct {
+		Type string `json:"@type"`
+		Key  string `json:"key"`
 	}
-	_, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
+	var pubKey PubKey
+	err = json.Unmarshal([]byte(jsonStr), &pubKey)
 	if err != nil {
-		return ErrInvalidProviderAddress
-	}
-	return nil
-}
-
-// NewMsgSetConsumerCommissionRate creates a new MsgSetConsumerCommissionRate msg instance.
-func NewMsgSetConsumerCommissionRate(consumerId string, commission math.LegacyDec, providerValidatorAddress sdk.ValAddress, signer string) *MsgSetConsumerCommissionRate {
-	return &MsgSetConsumerCommissionRate{
-		ConsumerId:   consumerId,
-		Rate:         commission,
-		ProviderAddr: providerValidatorAddress.String(),
-		Signer:       signer,
+		return "", "", err
 	}
+	return pubKey.Type, pubKey.Key, nil
 }
 
-func (msg MsgSetConsumerCommissionRate) Route() string {
-	return RouterKey
-}
-
-func (msg MsgSetConsumerCommissionRate) Type() string {
-	return TypeMsgSetConsumerCommissionRate
-}
-
-func (msg MsgSetConsumerCommissionRate) ValidateBasic() error {
-	if err := ValidateConsumerId(msg.ConsumerId); err != nil {
-		return err
-	}
-	_, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
-	if err != nil {
-		return ErrInvalidProviderAddress
-	}
-
-	if msg.Rate.IsNegative() || msg.Rate.GT(math.LegacyOneDec()) {
-		return errorsmod.Wrapf(ErrInvalidConsumerCommissionRate, "consumer commission rate should be in the range [0, 1]")
+// TODO create UT
+func ValidateTendermintHeader(header *ibctmtypes.Header) error {
```

Comment:
Validating a header involves substantially more things (e.g., check hashes, etc.). To avoid any confusion, maybe it would make sense to rename to something similar `HeaderContainsNonNilHeaderAndValidatorSet`.